renderers/src/main/java/com/pedrogomez/renderers/RendererBuilder.java

```diff
@@ -39,27 +44,125 @@
+public class RendererBuilder<T> {
 
-  private Collection<Renderer<T>> prototypes;
+  private List<Renderer<T>> prototypes;
 
   private T content;
   private View convertView;
   private ViewGroup parent;
   private LayoutInflater layoutInflater;
   private Integer viewType;
+  private Map<Class<T>, Class<? extends Renderer>> binding;
 
+  /**
+   * Initializes a RendererBuilder with an empty prototypes collection. Using this constructor some
+   * binding configuration is needed.
+   */
   public RendererBuilder() {
+    this(new LinkedList<Renderer<T>>());
+  }
 
+  /**
+   * Initializes a RendererBuilder with just one prototype. Using this constructor the prototype
+   * used will be always the same and the additional binding configuration wont be needed.
+   */
+  public RendererBuilder(Renderer<T> renderer) {
+    this(Collections.singletonList(renderer));
   }
 
-  public RendererBuilder(Collection<Renderer<T>> prototypes) {
-    if (prototypes == null || prototypes.isEmpty()) {
+  /**
+   * Initializes a RendererBuilder with a list of prototypes. Using this constructor some
+   * binding configuration is needed.
+   */
+  public RendererBuilder(List<Renderer<T>> prototypes) {
+    if (prototypes == null) {
+      throw new NeedsPrototypesException(
+          "RendererBuilder has to be created with a non null collection of"
+              + "Collection<Renderer<T> to provide new or recycled Renderer instances");
+    }
+    this.prototypes = prototypes;
+    this.binding = new HashMap<Class<T>, Class<? extends Renderer>>();
+  }
+
+  /**
+   * Get access to the prototypes collection used to create one RendererBuilder.
+   *
+   * @return prototypes collection.
+   */
+  public final Collection<Renderer<T>> getPrototypes() {
+    return prototypes;
+  }
+
+  /**
+   * Configure prototypes used as Renderer instances.
+   *
+   * @param prototypes to use by the builder in order to create Renderer instances.
+   */
+  public final void setPrototypes(List<Renderer<T>> prototypes) {
+    if (prototypes == null) {
+      throw new NeedsPrototypesException(
+          "RendererBuilder has to be created with a non null collection of"
+              + "Collection<Renderer<T> to provide new or recycled Renderer instances");
+    }
+    this.prototypes = prototypes;
+  }
+
+  /**
+   * Configure prototypes used as Renderer instances.
+   *
+   * @param prototypes to use by the builder in order to create Renderer instances.
+   * @return the current RendererBuilder instance.
+   */
+  public RendererBuilder<T> withPrototypes(List<Renderer<T>> prototypes) {
+    if (prototypes == null) {
       throw new NeedsPrototypesException(
-          "RendererBuilder have to be created with a non empty collection of"
+          "RendererBuilder has to be created with a non null collection of"
               + "Collection<Renderer<T> to provide new or recycled Renderer instances");
     }
     this.prototypes = prototypes;
```

Comment:
```
builder.withPrototypes(...).withPrototypes(...)
```

This snippet will remove old prototypes, right? Does it make sense to use ``addAll`` instead?